src/main/kotlin/org/dif/crypto/key/RecipientKeySelector.kt

```diff
@@ -1,24 +1,54 @@
 package org.dif.crypto.key
 
+import com.nimbusds.jose.jwk.Curve
 import org.dif.diddoc.DIDDocResolver
 import org.dif.exceptions.DIDDocException
 import org.dif.exceptions.DIDDocNotResolvedException
+import org.dif.exceptions.IncompatibleCryptoException
+import org.dif.exceptions.SecretNotFoundException
 import org.dif.secret.SecretResolver
 import org.dif.utils.divideDIDFragment
 import org.dif.utils.isDIDFragment
 
 class RecipientKeySelector(private val didDocResolver: DIDDocResolver, private val secretResolver: SecretResolver) {
-    fun verifyKey(selector: String): Key = Key.wrapVerificationMethod(
+    fun findVerificationKey(signFrom: String): Key = Key.wrapVerificationMethod(
         let {
-            val (did) = divideDIDFragment(selector)
-            val didDoc = didDocResolver.resolve(did).orElseThrow { throw DIDDocNotResolvedException(did) }
+            check(isDIDFragment(signFrom)) { "'DID URL' was excepted" }
 
-            val verificationMethodId = if (isDIDFragment(selector)) { selector } else {
-                didDoc.authentications.firstOrNull()
-                    ?: throw DIDDocException("Authentication is not found in DID Doc '$did'")
+            val (did) = divideDIDFragment(signFrom)
+            didDocResolver.resolve(did)
+                .map { it.findVerificationMethod(signFrom) }
+                .orElseThrow { throw DIDDocNotResolvedException(did) }
+        }
+    )
+
+    fun findAuthCryptKeys(from: String, to: List<String>): Pair<Key, Sequence<Key>> {
+        check(isDIDFragment(from)) { "'DID URL' was excepted" }
+
+        val (did) = divideDIDFragment(from)
+        return didDocResolver.resolve(did)
+            .map { it.findVerificationMethod(from) }
+            .map { Key.wrapVerificationMethod(it) }
+            .map { Pair(it, findRecipientKeys(to, it.curve)) }
+            .orElseThrow { DIDDocNotResolvedException(did) }
```

Comment:
I believe two separate exceptions need to be thrown here: `DIDDocNotResolvedException` if DID DOc is not resolved (null), and `DIDUrlNotFoundError` if a key is not found within the DID Doc (see https://github.com/sicpa-dlab/didcomm-python/blob/main/didcomm/errors.py)